cgi-bin/DW/Media/Photo.pm

```diff
@@ -174,4 +174,45 @@ sub _select_version {
     # the resized image.
 }
 
+# this adds on to the base method by also deleting any associated thumbnails
+sub delete {
+    my $self = $_[0];
+    my $deleted = $self->SUPER::delete;
+    return 0 unless $deleted;  # was already deleted
+
+    # at this point the image has just been deleted - look for thumbnails
+    my $u = $self->u
+        or croak 'Sorry, unable to load the user.';
+    my @mv = $u->selectrow_array( "SELECT mediaid FROM media_versions" .
+                                  " WHERE userid=? AND versionid=?", undef,
+                                  $u->id, $self->versionid );
+    # this shouldn't happen unless something was saved in an incomplete state
+    # or the database has taken an unexpected vacation
+    return $deleted unless @mv;
+    my ( $mediaid ) = @mv;
+
+    if ( $mediaid == $self->versionid ) {
```

Comment:
Prefer not to indent large sections of code. In this case, this would be nice:

```
    # not an original, so no further work
    return $deleted unless $mediaid == $self->versionid;

    # was an original, do work...
```